### PR TITLE
map gpu in kube tasks

### PR DESCRIPTION
**Pull request by xudifsd**
Tasks run by nvidia-container-runtime will have ENV of `NVIDIA_VISIBLE_DEVICES` and value is GPU uuids, these uuid can mapped into minor number by info get from nvidia-smi.

This PR will make job-exporter able to get resource usage of tasks run by either yarn-launcher and kube-launcher, including GPU usage.

### CODE REVIEW

File: src/job-exporter/src/collector.py [hunk 1]
```diff
@@ -470,20 +473,27 @@ def collect_impl(self):
...
+
+        if inspect_info.gpu_ids:
+            ids = inspect_info.gpu_ids.replace("\"", "").split(",")
+            for id in ids:
+                if id.isdigit():
+                    gpu_ids.append(id)
+                elif id and gpu_infos is not None:
+                    # id is in form of UUID like
+                    # GPU-dc0671b0-61a4-443e-f456-f8fa6359b788
+                    if gpu_infos.get(id) is not None:
```
Comment (yqwang-ms) [inline src/job-exporter/src/collector.py]: Do we need a warning log if it is None?

File: src/job-exporter/src/collector.py [hunk 2]
```diff
@@ -470,20 +473,27 @@ def collect_impl(self):
...
+
+        result_labels["username"] = inspect_info.username or "unknown"
+        result_labels["job_name"] = inspect_info.job_name or "unknown"
+        result_labels["role_name"] = inspect_info.role_name or "unknown"
+        result_labels["task_index"] = inspect_info.task_index or "unknown"
+
+        if inspect_info.gpu_ids:
+            ids = inspect_info.gpu_ids.replace("\"", "").split(",")
+            for id in ids:
+                if id.isdigit():
```
Comment (yqwang-ms) [inline src/job-exporter/src/collector.py]: In which case, it is directlt the gpu id, and in which case it is just a UUID?

**Comment by scarlett2018** [comment]
TODO: help check regression during release testing process
@xudifsd

File: src/job-exporter/src/collector.py (hunk 1 above)
Comment (xudifsd) [reply src/job-exporter/src/collector.py]: added

File: src/job-exporter/src/collector.py (hunk 2 above)
Comment (xudifsd) [reply src/job-exporter/src/collector.py]: added comment in code.

Containers from yarn will have minor in gpu id, which are digits.

Containers from kube launcher will have UUID in gpu id.

**Comment by xudifsd** [comment]
test pass in release process

**Pull request merged**